frontend/packages/dev-console/src/components/import/image-search/ImageSearch.tsx

```diff
@@ -17,98 +17,93 @@ import { secretModalLauncher } from '../CreateSecretModal';
+        .then((imageStreamImport) => {
+          const status = _.get(imageStreamImport, 'status.images[0].status');
+          if (status.status === 'Success') {
+            const name = _.get(imageStreamImport, 'spec.images[0].from.name');
+            const image = _.get(imageStreamImport, 'status.images[0].image');
+            const tag = _.get(imageStreamImport, 'status.images[0].tag');
+            const isi = { name, image, tag, status };
+            const ports = getPorts(isi);
+            setFieldValue('isSearchingForImage', false);
+            setFieldValue('isi.name', name);
+            setFieldValue('isi.image', image);
+            setFieldValue('isi.tag', tag);
+            setFieldValue('isi.status', status);
+            setFieldValue('isi.ports', ports);
+            setFieldValue('image.ports', ports);
+            setFieldValue('image.tag', tag);
+            !values.name && setFieldValue('name', getSuggestedName(name));
+            !values.application.name &&
+              values.application.selectedKey !== UNASSIGNED_KEY &&
+              setFieldValue('application.name', `${getSuggestedName(name)}-app`);
+            // set default port value
+            const targetPort =
+              (!initialValues.route.targetPort || touched.searchTerm) && _.head(ports);
+            targetPort && setFieldValue('route.targetPort', makePortName(targetPort));
+            setValidated(ValidatedOptions.success);
+          } else {
+            setFieldValue('isSearchingForImage', false);
+            setFieldValue('isi', {});
+            setFieldValue('isi.status', status);
+            setFieldValue('route.targetPort', null);
+            setValidated(ValidatedOptions.error);
+          }
+        })
+        .catch((error) => {
           setFieldValue('isi', {});
-          setFieldValue('isi.status', status);
-          setFieldValue('route.targetPort', null);
+          setFieldValue('isi.status', { metadata: {}, status: '', message: error.message });
+          setFieldValue('isSearchingForImage', false);
           setValidated(ValidatedOptions.error);
-        }
-      })
-      .catch((error) => {
-        setFieldValue('isi', {});
-        setFieldValue('isi.status', { metadata: {}, status: '', message: error.message });
-        setFieldValue('isSearchingForImage', false);
-        setValidated(ValidatedOptions.error);
-      });
-  }, [
-    setFieldValue,
-    values.application.name,
-    values.application.selectedKey,
-    values.name,
-    values.project.name,
-    values.searchTerm,
-    touched,
-    initialValues.route.targetPort,
-  ]);
+        });
+    }, 500),
+    [
+      setFieldValue,
+      touched,
+      values.application.name,
+      values.application.selectedKey,
+      values.name,
+      values.project.name,
+      initialValues.route.targetPort,
+    ],
+  );
 
   const handleSave = (name: string) => {
     setNewImageSecret(name);
-    values.searchTerm && handleSearch();
```

Comment:
I think this one is being used to trigger the validation again when a pull secret is added. So we need this one as well.